helix-view/src/document.rs

```diff
@@ -2182,10 +2187,19 @@ impl Document {
             }
         }
 
-        match &self.language {
-            Some(lang) => lang.as_ref().auto_pairs.as_ref().or(global_config),
-            None => global_config,
-        }
+        let selection = self.selection(view.id).primary();
+        self.syntax
+            .as_ref()
+            .and_then(|syntax| {
+                syntax
+                    .layers_for_byte_range(selection.from() as u32, selection.to() as u32)
+                    .last()
```

Comment:
`Syntax::layer_for_byte_range` does the same thing here, it returns the last (i.e. smallest) injection covering the byte range. For this PR we can reuse `layer_for_byte_range` to make this a little bit shorter. We will want `layers_for_byte_range` for other changes though.